### ISSUE TITLE
multiclass segmentation for aerial dataset

### ISSUE THREAD

**Issue by amardeepjaiman** [Oslandia/mapillary_classification#142]
Hi,

is it possible to use the aerial dataset implementation for multiclass dataset by changing the number of input classes etc ? any lead ?

**Comment by delhomer** [comment]
Hello @amardeepjaiman ,

You certainly can! A direct example is provided by the Tanzania dataset, that encompasses 3 classes (finished buildings, incomplete buildings, foundations).

From what we saw, a postprocessing step is needed to clean the semantic segmentation algorithm output (it may predict mixed-classes buildings). As an alternative you may consider alternative models in order to do your prediction.

**Comment by amardeepjaiman** [comment]
Thanks @delhomer . One more clarification I needed, From the source code it looks like that, a custom Unet is implemented in this project. Can we somehow change the encoder backbone as resnet 34 or resnet50 or anything else in the current source code ? So that i can load imagenet weights available for resnet architectures


Please suggest.

**Comment by delhomer** [comment]
As far as I know, it is possible but not straightforward. You should know the exact composition of available Resnet architectures (layers, batch normalization, ... ) in order to assign pre-loaded weights to your own model. If there is any misaligned layer, it will fail. The simplest thing to do is probably to build a `Model` (as declared in this project) by calling freed architectures (like `keras.applications.resnet50.Resnet50()`).

We did this project as a Proof-of-Concept several months ago, without knowing such implementations. It will be hard (if not impossible) for me to spend time to include this implementation into the project now, hence do not hesitate to contribute by submitting PRs if you succeed in that. Even if you do not contribute, feel free to provide a feedback of your attempt here. :)
